File: test_bitset.java
import java.util.BitSet;

class test_bitset
{
  private static void print(BitSet bits)
  {
    for (int i = 0; i < bits.size(); i++)
      System.out.print(bits.get(i) ? "1" : "0");
    System.out.println();
  }
  public static void main(String[] args)
  {
    BitSet bits = new BitSet();
    print(bits);
    for (int i = 0; i < 16; i++)
      bits.set(i*4);
    bits.set(100);
    print(bits);
  }
}

File: test_bitset.cpp
#include <bitset>
#include <iostream>
using namespace std;

void change(bitset<64>* bits)
{
  for (int i = 0; i < 16; i++)
    bits->reset(i*4);
  bits->reset(100);
}

int main()
{
  bitset<64> bits(0xFFFFFFFFFFFFFFFF);
  cout << "bits: " << bits << endl;
  change(&bits);
  cout << "bits: " << bits << endl;
}
